### PR TITLE
Make `-r` option work (#87)

### PR DESCRIPTION
**Pull request by irushchyshyn**
- Introduce `%{srcname}` macro to be used alongside `%{pypi_name}` if the  package name is to be changed (applies also for packages already starting with `python-` or `Python-` in PyPI)
- Fix filters tests to work with the new signature for `macroed_pkg_name`
- Add a new integration test for `python-sphinx.spec`
- Corner case: fix searching for Sphinx directory, when it is tested itself (another `doc/conf.py` in tests folder)

Fixes issues #87 and #66.

### CODE REVIEW

File: pyp2rpm/templates/epel7.spec [hunk 1]
```diff
@@ -75,7 +78,7 @@ ln -sf %{_bindir}/{{ script|script_name_for_python_version(pv) }} %{buildroot}/%
 {%- endfor %}
 {%- endif %}
 {% for pv in [data.base_python_version] + data.python_versions %}
-%files -n {{ data.pkg_name|macroed_pkg_name(data.name)|name_for_python_version(pv, True, True) }} 
+%files -n {{ data.pkg_name|macroed_pkg_name(data.srcname)|name_for_python_version(pv, True, True) }}
```
Comment (mcyprian) [inline pyp2rpm/templates/epel7.spec]: Trailing whitespace present at the end of this line.

File: pyp2rpm/templates/fedora.spec [hunk 2]
```diff
@@ -74,7 +77,7 @@ ln -sf %{_bindir}/{{ script|script_name_for_python_version(pv) }} %{buildroot}/%
 {%- endfor %}
 {%- endif %}
 {% for pv in [data.base_python_version] + data.python_versions %}
-%files -n {{ data.pkg_name|macroed_pkg_name(data.name)|name_for_python_version(pv, True) }} 
+%files -n {{ data.pkg_name|macroed_pkg_name(data.srcname)|name_for_python_version(pv, True) }}
```
Comment (mcyprian) [inline pyp2rpm/templates/fedora.spec]: Trailing whitespace present at the end of this line.

**Review by mcyprian — COMMENTED** [review]
This PR fixes also  #66, right? Could you reference it in one of the commits? 
It seems to work fine for both cases:
    - name of the package is explicitly changed by -r option
    - original PyPI name contains python- prefix

**Comment by mcyprian** [comment]
There is also couple of packages containing "Python-" prefix on PyPI. Python-Crest for example, in this case result is following:
```
%global pypi_name Python-Crest

Name:           python-%{pypi_name}
```
Maybe we should cover this case as well.

File: pyp2rpm/templates/epel7.spec (hunk 1 above)
Comment (irushchyshyn) [reply pyp2rpm/templates/epel7.spec]: Fixed in 2c7783f710a97c4e8dbb1bf29ce35327aaea86c0

File: pyp2rpm/templates/fedora.spec (hunk 2 above)
Comment (irushchyshyn) [reply pyp2rpm/templates/fedora.spec]: Fixed in 2c7783f710a97c4e8dbb1bf29ce35327aaea86c0

**Comment by irushchyshyn** [comment]
> There is also couple of packages containing "Python-" prefix on PyPI. Python-Crest for example, in this case result is following:
> %global pypi_name Python-Crest
> Name:                    python-%{pypi_name}
> Maybe we should cover this case as well.

Thanks for a suggestion. I have covered those packages in 2c7783f

**Comment by irushchyshyn** [comment]
> This PR fixes also #66, right? Could you reference it in one of the commits?
> It seems to work fine for both cases:
> - name of the package is explicitly changed by -r option
> - original PyPI name contains python- prefix

Thanks for a review. I have referenced the issue in the PR description.

**Comment by mcyprian** [comment]
Works like a charm :)
